javaparser-core/src/main/java/com/github/javaparser/ast/CompilationUnit.java

```diff
@@ -505,17 +505,6 @@ public AnnotationDeclaration addAnnotationDeclaration(String name, Modifier.Keyw
     public Optional<ClassOrInterfaceDeclaration> getClassByName(String className) {
         return getTypes().stream().filter(type -> type.getNameAsString().equals(className) && type instanceof ClassOrInterfaceDeclaration && !((ClassOrInterfaceDeclaration) type).isInterface()).findFirst().map(t -> (ClassOrInterfaceDeclaration) t);
     }
-    
-    /**
-     * Try to get all local class declarations ending by its name (top level or inner class)
-     *
-     * @param className the class name (case-sensitive)
-     */
-    public List<ClassOrInterfaceDeclaration> getLocalDeclarationFromClassname(String className) {
```

Comment:
Removing this public method is an API change -- is there a way to avoid removing it?

Preferably we would deprecate the method first, with javadoc to explain that it does not work in some cases.

Reply:
Of course no problem with it, I thought that it has no purpose anymore.

Reply:
Where there's an alternative preferred way to do it, deprecate it and explain what the alternative is please :) 

(perhaps rewrite the content of the method to use the alternative technique too)

Reply:
Ok, no problem. So I will revert the commits with method deletion, and add javadoc to the new method.